CommonLibrary.cmake

```diff
@@ -28,6 +28,66 @@ set(COMMON_LIBRARY_TYPE SHARED CACHE STRING
   "Library type {any combination of SHARED, STATIC}")
 set_property(CACHE COMMON_LIBRARY_TYPE PROPERTY STRINGS SHARED STATIC)
 
+#-------------------------------------------------------------------------------
+# a macro which will add _D to debug library names on WIN32 builds
+# called by common_library() defined below.
+# has no effect on linux/mac unless the user sets COMMON_LIBRARY_DEBUG_SUFFIX=ON
+#-------------------------------------------------------------------------------
+MACRO (COMMON_SET_LIB_NAME_OPTIONS libtarget libname libtype)
+  # message (STATUS "${libname} libtype: ${libtype}")
+  if (${libtype} MATCHES "SHARED")
+    # default for linux/mac/unix
+    set (LIB_RELEASE_NAME "lib${libname}")
+    set (LIB_DEBUG_NAME "lib${libname}")
+    if (WIN32)
+      set (LIB_RELEASE_NAME "${libname}")
+      set (LIB_DEBUG_NAME "${libname}_D")
+    elseif (COMMON_LIBRARY_DEBUG_SUFFIX) 
+      set (LIB_RELEASE_NAME "${libname}")
```

Comment:
Unless I'm mistaken, this would generate unix libs without the lib prefix, is this intended?